arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/continuations/EagerEffectSpec.kt

```diff
@@ -28,13 +30,13 @@ class EagerEffectSpec : StringSpec({
       }.fold({ fail("Should never come here") }, ::identity) shouldBe i
     }
   }
-
+  
   "try/catch - finally works" {
     checkAll(Arb.string(), Arb.int()) { s, i ->
       val promise = CompletableDeferred<Int>()
-      eagerEffect<String, Int> {
+      eagerEffect {
         try {
-          shift(s)
+          shift<Int>(s)
```

Comment:
Perhaps we should explore after this PR if `shift` can be implemented by returning `Nothing` and if it brings better or worse ergonomics / type inference

Reply:
Agreed, that would be a good thing to explore. I found the argument that `fun error` returns `Nothing` very compelling. Comparing it to `emptyList` was probably incorrect.